lib/virt_autotest/utils.pm

```diff
@@ -607,4 +607,50 @@ sub check_port_state {
     return $port_state;
 }
 
+=head2 subscribe_extensions_and_modules
+
+  subscribe_extensions_and_modules(dst_machine => $machine, activate => 1/0, reg_exts => $exts)
+
+Any available extensions and modules listed out by SUSEConnect --list-extensions
+that do not require additional regcode can be subscribe directly by using command
+SUSEConnect -p [extension or module]. Subscription is to be performed on localhost
+by default if argument dst_machine is not given any other address. Deactivation is
+also supported if argument activate is given 0 explicitly. Multiple extensions or
+modules can be passed in as a single string separated by space to argument reg_exts
+to be subscribed one by one.
+
+=cut
+sub subscribe_extensions_and_modules {
+    my (%args) = @_;
+    $args{dst_machine} //= 'localhost';
+    $args{activate} //= 1;
+    $args{reg_exts} //= '';
+    croak('Nothing to be subscribed. Please pass something to argument reg_exts.') if ($args{reg_exts} eq '');
+
+    my $cmd = '';
+    $cmd = "SUSEConnect -l";
+    $cmd = "ssh root\@$args{dst_machine} " . "\"$cmd\"" if ($args{dst_machine} ne 'localhost');
+    my $ret = script_run($cmd);
+    save_screenshot;
+    unless ($ret == 0) {
+        record_info("Base product not registered or no extensions/modules available.", script_output($cmd, proceed_on_failure => 1));
```

Comment:
It could fail by connection problem to dst_machine, or password needed for ssh command if the dst_machine is not in authorized machine list?

Reply:
You should setup your connection and make sure it works. This is not part of this subroutine.

Reply:
Okay, so different prerequisites. Would you please add a comment for this in the function description?

Reply:
Done